thoth/python/source.py

```diff
@@ -323,12 +372,20 @@ def provides_package_version(self, package_name: str, package_version: str) -> b
             return False
 
     @lru_cache(maxsize=10)
-    def get_package_hashes(self, package_name: str, package_version: str) -> list:
+    def get_package_hashes(self, package_name: str, package_version: str, with_included_files: bool = False) -> list:
         """Get information about release hashes available in this source index."""
         if self.warehouse:
-            return self._warehouse_get_package_hashes(package_name, package_version)
+            return self._warehouse_get_package_hashes(package_name, package_version, with_included_files)
+
+        artifacts_sha = self._download_artifacts_sha(package_name, package_version, with_included_files)
+        result = []
+        for artifact_item in artifacts_sha:
+            print(len(artifact_item))
+            doc = {}
+            doc['name'] = artifact_item[0]
+            doc['sha256'] = artifact_item[1]
+            if with_included_files:
+                doc['contents'] = artifact_item[2]
```

Comment:
I am also wondering would it be good if we could check the len before we use it?

Reply:
@fridex what do you think? I think like if there is a flag with_included_files, then there needs to be third data which contents